Read build info from package.json

mhofman opened the pull request:
Follow-up to #52, but instead of relying on new commands and positional parameters, the base build info is read from the `package.json` enclosing the `proposals`. This allows using a single and simple `synthetic-chain build` command, whether in a3p itself or in other repos integrations. It also provides a place in the future to configure the build of other a3p like images.

I have manually locally tested this change with https://github.com/Agoric/agoric-sdk/pull/8743.

A new version of the `synthetic-chain` package still needs to be published if approved.

turadg commented on packages/synthetic-chain/cli.ts:
```diff
@@ -20,7 +21,9 @@ const { positionals, values } = parseArgs({
 });
 
-const allProposals = readProposals(path.resolve('.'));
+const proposalsParent = path.resolve('.');
```
this isn't inaccurate but I think more importantly it's the root of the testing project.
```suggestion
const root = path.resolve('.');
```

turadg reviewed (changes requested):
I like where this is going but I think this increment is a little more confusing on the way to something better.

If we want the package.json to control the build config, then let's go all the way. 

This one would be:
```js
    "agoricSyntheticChain": {
        "mode": "fromAg0",
    },
```

Then appending to the default (main) would be:
```js
    "agoricSyntheticChain": {
        "mode": "append",
    },
```

And a specific branch would be:
```js
    "agoricSyntheticChain": {
        "mode": "append",
        "fromTag": "my-tag",
    },
```

Then the CLI command could be simply `synthetic-chain build` for all cases.

mhofman commented on packages/synthetic-chain/cli.ts:
```diff
@@ -49,16 +52,19 @@ const buildImages = () => {
   case 'amend': // alias for backcompat
-  case 'append':
-    const fromTag = positionals[1] || baseTag;
+  case 'append': { // alias for backcompat
```
I wanted to deprecate the `append` command line as well, and keep only `build`

mhofman commented:
I clarified that using `null` is the toggle to build from scratch, and I removed the extra commands to only keep `build`

turadg commented on packages/synthetic-chain/upgrade-test-scripts/start_ag0.sh:
```diff
@@ -17,6 +17,7 @@ sed -i.bak "s/^enable-unsafe-cors =.*/enable-unsafe-cors = true/" "$HOME/.agoric
 sed -i.bak "s/127.0.0.1:26657/0.0.0.0:26657/" "$HOME/.agoric/config/config.toml"
 sed -i.bak "s/cors_allowed_origins = \[\]/cors_allowed_origins = \[\"*\"\]/" "$HOME/.agoric/config/config.toml"
 sed -i.bak '/^\[api]/,/^\[/{s/^enable[[:space:]]*=.*/enable = true/}' "$HOME/.agoric/config/app.toml"
+sed -i.bak "s/^pruning =.*/pruning = \"nothing\"/" "$HOME/.agoric/config/app.toml"
```
this should be its own commit but I'll approve to get this in faster.

turadg reviewed (approved):
Nice improvement

turadg commented:
Failure was `Unable to find image 'ghcr.io/agoric/agoric-3-proposals:test-upgrade-13' locally`, which is a deficiency of the build daemon that it dropped from build results.

Since it has to run again, I moved the prune change into its own commit and added the version release commit.

mhofman commented:
> Since it has to run again, I moved the prune change into its own commit and added the version release commit.

Oh my bad I didn't mean to include that in this PR, it was a local experiment. But I guess it's there now

turadg commented:
Released as https://www.npmjs.com/package/@agoric/synthetic-chain/v/0.0.2
